### PR TITLE
DBZ-5075: Clean up MySQL connector codebase

### PR DESCRIPTION
**Pull request by morozov**
1. `DdlChanges#groupStatementStringsByDatabase()` and `DatabaseStatementStringConsumer` interface were introduced in #49 but weren't used anywhere since then.
2. `DdlChanges#groupStatementsByDatabase`, `DatabaseStatementConsumer` and `HaltingPredicate` are unused since #3373.
3. `MySqlSchema#applyDdl()` is also no longer available.
4. The only terminator used in `DdlChanges` is a semicolon.

### CODE REVIEW

File: debezium-connector-mysql/src/test/java/io/debezium/relational/ddl/DdlChangesTest.java [hunk 1]
```diff
@@ -1,97 +0,0 @@
-/*
```
Comment (jpechane) [inline debezium-connector-mysql/src/test/java/io/debezium/relational/ddl/DdlChangesTest.java:1]: Why is this test completely removed?

**Comment by jpechane** [comment]
@morozov Hi, nice initiative! Overall LGTM, I've just dropped a single question.

File: debezium-connector-mysql/src/test/java/io/debezium/relational/ddl/DdlChangesTest.java (hunk 1 above)
Comment (morozov) [reply debezium-connector-mysql/src/test/java/io/debezium/relational/ddl/DdlChangesTest.java:1]: All its methods test `DdlChanges#groupEventsByDatabase()` which isn't used anywhere else than in this test.

**Comment by jpechane** [comment]
@morozov Applied, go for more chopping :-)

**Pull request merged**